packages/ra-core/src/form/ValidationError.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { FunctionComponent } from 'react';
+import {
+    ValidationErrorMessage,
+    ValidationErrorMessageWithArgs,
+} from './validate';
+import { useTranslate } from '../i18n';
+import { Translate } from '../types';
+
+interface Props {
+    error: ValidationErrorMessage;
+}
+
+const translateArrayArgValue = (translate: Translate, values: any[]) =>
+    values
+        .map(value => translate(value.toString(), { _: value.toString() }))
+        .join(', ');
+
+const translateArgValue = (translate: Translate, argValue: any) => {
+    if (Array.isArray(argValue)) {
+        return translateArrayArgValue(translate, argValue);
+    }
+
+    return translate(argValue.toString(), { _: argValue.toString() });
+};
+
+const translateArgs = (translate: Translate, args: { [key: string]: any }) =>
```

Comment:
I don't understand what you're doing here. We should not translate arguments. we should translate the message.